### PR TITLE
Migrated to husky v7 and added a commitlint with GHA check for each PR

### PR DESCRIPTION
**Pull request by aleks-elkin**
- Migrated husky from v4 to v7 (there are some breaking changes).
- Added a commitlint package to validate commit messages with conventional commits standard. It will be used later to make automatic semantic releases based on conventional changelog with lerna. Documentation about the format: https://www.conventionalcommits.org/en/v1.0.0/
- Added a GHA to automatically check commit messages in each PR.

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+npm run update-readme && npm run format && npx lint-staged && npm run check-deps
```
Comment (wejendorp) [inline .husky/pre-commit:4]: Wont this be a bit much? Or is it quite fast to run those "all files" hooks?

File: package.json [hunk 2]
```diff
@@ -67,12 +67,8 @@
...
     "update-readme": "cross-env-shell node ./scripts/update-readme-files.js",
     "add-desc-to-src": "cross-env-shell node ./scripts/add-description-to-src.js",
-    "watch": "cross-env-shell lerna exec --parallel -- rollup -w -c=`pwd`/rollup.config.js"
-  },
-  "husky": {
-    "hooks": {
-      "pre-commit": "npm run update-readme && lint-staged && npm run check-deps"
-    }
+    "watch": "cross-env-shell lerna exec --parallel -- rollup -w -c=`pwd`/rollup.config.js",
+    "prepare": "husky install"
```
Comment (wejendorp) [inline package.json:71]: Why did we choose `prepare`? Seems it's only relevant on `postinstall`, not on publish etc

**Review by wejendorp — APPROVED** [review]
Consider the comments, but nice, go ahead :)

File: .husky/pre-commit (hunk 1 above)
Comment (aleks-elkin) [reply .husky/pre-commit:4]: Well, all of them where here before, I just move them to new config file for husky.
On my machine it took several seconds to complete. It is noticeable, but not annoying, IMO.

File: package.json (hunk 2 above)
Comment (aleks-elkin) [reply package.json:71]: It was done by husky migration tool. I think here you could find the explanation behind this decision:
https://github.com/typicode/husky/issues/884

Comment (wejendorp) [reply package.json:71]: Great, thanks :)

> Unlike postinstall, the prepare command does not run when a NPM package is installed as a dependency in another project.

That's reason enough :+1:

**Pull request merged**